master/pkg/stream/stream.go

```diff
@@ -109,15 +147,23 @@ func (p *Publisher[T]) Broadcast(events []Event[T]) {
 
 	// check each event against each subscription
 	for _, sub := range p.Subscriptions {
-		func(){
+		func() {
 			for _, ev := range events {
-				var msg *websocket.PreparedMessage
+				var msg interface{}
 				if ev.After != nil && sub.filter(*ev.After) {
+					if ev.upsertCache == nil {
+						ev.upsertCache = sub.Streamer.PrepareFn(ev.After)
```

Comment:
Wait... who makes the UpsertMsg?

Reply:
ah my bad I missed this before, but I see now I need to refactor event as well - the upsertCache and deleteCache shouldn't be interfaces anymore